addons/main/functions/VehicleAction/fnc_doSelectHEMunition.sqf

```diff
@@ -0,0 +1,99 @@
+#include "script_component.hpp"
+/*
+ * Author: diwako
+ * Makes main gun of vehicle switch to HE ammo,
+ * then switches back to regular ammo after 15 seconds
+ *
+ * Arguments:
+ * 0: vehicle suppressing <OBJECT>
+ *
+ * Return Value:
+ * success
+ *
+ * Example:
+ * [vehicle bob] call lambs_main_fnc_doSelectHEMunition;
+ *
+ * Public: No
+*/
+params ["_vehicle"];
+private _pos = getPosATL player;
+private _gunner = gunner _vehicle;
+
+if (isNull _gunner) exitWith {};
+
+// figure out turret and ammo
+private _turretPath = (assignedVehicleRole _gunner) select 1;
+private _turretMagazines = _vehicle magazinesTurret _turretPath;
+private _vehicleMagazines = magazines [_vehicle, false];
+private _turrets = _vehicle weaponsTurret _turretPath;
+
+// kick out magazines that do not belong to the gunner turrets
+private _validMags = _turretMagazines arrayIntersect _vehicleMagazines;
+
+private _turret = "";
+private _muzzle = "";
+private _heMag = "";
+
+{
+    _turret = _x;
+    private _muzzles = getArray (configFile >> "CfgWeapons" >> _turret >> "muzzles");
+
+    // first pass, try to find the "HE" muzzle
+    private _index = _muzzles findIf {_x == "HE"};
+    if (_index > -1) then {
+        _muzzle = _muzzles select _index;
+        break;
+    };
+
+    // second pass no "HE" muzzle not found, see if any ammo loaded has its warheadName values set as "HE"
+    _index = _muzzles findIf {
+        {
+            private _ammo = getText (configFile >> "CfgMagazines" >> _x >> "ammo");
+            if (_ammo isEqualTo "") then {continue};
+            if ((getText (configFile >> "CfgAmmo" >> _ammo >> "warheadName")) == "HE") exitWith {
+                _heMag = _x;
+            };
+        } forEach (if (_x == "this") then {
+            private _arr = _validMags arrayIntersect getArray ((configFile >> "CfgWeapons" >> _turret >> "magazines"));
+            reverse _arr;
+            _arr
+        } else {
+            private _arr = _validMags arrayIntersect getArray ((configFile >> "CfgWeapons" >> _turret >> _x >> "magazines"));
+            reverse _arr;
+            _arr
+        });
```

Comment:
i dont like this code part here. i feel like this should be done before the if and than just use the _arr variable in the foreach.